packages/block-library/src/embed/edit.js

```diff
@@ -0,0 +1,382 @@
+		 * @return {Object} Attributes and values.
+		 */
+		getAttributesFromPreview( preview, allowResponsive = true ) {
+			const attributes = {};
+			// Some plugins only return HTML with no type info, so default this to 'rich'.
+			let { type = 'rich' } = preview;
+			// If we got a provider name from the API, use it for the slug, otherwise we use the title,
+			// because not all embed code gives us a provider name.
+			const { html, provider_name: providerName } = preview;
+			const providerNameSlug = kebabCase( toLower( '' !== providerName ? providerName : title ) );
+
+			if ( isFromWordPress( html ) ) {
+				type = 'wp-embed';
+			}
+
+			if ( html || 'photo' === type ) {
+				attributes.type = type;
+				attributes.providerNameSlug = providerNameSlug;
+			}
+
+			attributes.className = classnames(
+				this.props.attributes.className,
+				this.getAspectRatioClassNames( html, allowResponsive )
+			);
+
+			return attributes;
+		}
+
+		/***
+		 * Sets block attributes based on the preview data.
+		 */
+		setAttributesFromPreview() {
+			const { setAttributes, preview } = this.props;
+			const { allowResponsive } = this.props.attributes;
+			setAttributes( this.getAttributesFromPreview( preview, allowResponsive ) );
+		}
+
+		switchBackToURLInput() {
+			this.setState( { editingURL: true } );
+		}
+
+		getResponsiveHelp( checked ) {
+			return checked ? __( 'Videos and other content automatically resizes.' ) : __( 'Content is fixed size.' );
+		}
+
+		toggleResponsive() {
+			const { allowResponsive, className } = this.props.attributes;
+			const { html } = this.props.preview;
+			const responsiveClassNames = this.getAspectRatioClassNames( html, ! allowResponsive );
+
+			this.props.setAttributes(
+				{
+					allowResponsive: ! allowResponsive,
+					className: classnames( className, responsiveClassNames ),
+				}
+			);
+		}
+
+		render() {
+			const { url, editingURL } = this.state;
+			const { caption, type, allowResponsive } = this.props.attributes;
+			const { fetching, setAttributes, isSelected, className, preview, cannotEmbed, supportsResponsive } = this.props;
+			const controls = (
+				<Fragment>
+					<BlockControls>
+						<Toolbar>
+							{ preview && ! cannotEmbed && (
+								<IconButton
+									className="components-toolbar__control"
+									label={ __( 'Edit URL' ) }
+									icon="edit"
+									onClick={ this.switchBackToURLInput }
+								/>
+							) }
+						</Toolbar>
+					</BlockControls>
+					{ supportsResponsive && (
+						<InspectorControls>
+							<PanelBody title={ __( 'Media Settings' ) } className="blocks-responsive">
+								<ToggleControl
+									label={ __( 'Automatically scale content' ) }
+									checked={ allowResponsive }
+									help={ this.getResponsiveHelp }
+									onChange={ this.toggleResponsive }
+								/>
+							</PanelBody>
+						</InspectorControls>
+					) }
+				</Fragment>
+			);
+
+			if ( fetching ) {
```

Comment:
There's no need to do the overhead of building the controls if `fetching===true`, this could be moved higher in the render function